homeassistant/config.py

```diff
@@ -427,19 +427,35 @@ def async_log_exception(ex: vol.Invalid, domain: str, config: Dict,
 
 
 @callback
-def _format_config_error(ex: vol.Invalid, domain: str, config: Dict) -> str:
+def _format_config_error(ex: vol.Invalid, domain: str, config: Dict,
+                         in_multiple_invalid: bool = False) -> str:
     """Generate log exception for configuration validation.
 
     This method must be run in the event loop.
     """
-    message = "Invalid config for [{}]: ".format(domain)
+    message = "" if in_multiple_invalid else \
+        "Invalid config for [{}]: ".format(domain)
+
+    if isinstance(ex, vol.MultipleInvalid):
+        # Multiple validation errors. Let's print all of them
+        return message + '\n'.join(sorted(
+            _format_config_error(sub_error, domain, config,
+                                 in_multiple_invalid=True)
+            for sub_error in ex.errors
+        ))
+
     if 'extra keys not allowed' in ex.error_message:
-        message += '[{}] is an invalid option for [{}]. Check: {}->{}.'\
-                   .format(ex.path[-1], domain, domain,
-                           '->'.join(str(m) for m in ex.path))
+        message += '[{}] is an invalid option for [{}].'\
+                   .format(ex.path[-1], domain)
+    elif u'required key not provided' in ex.error_message:
```

Comment:
Strings are unicode by default in Python 3.

Reply:
Thanks! That's a copy-paste error from esphomeyaml 😁 (stuck on py2 there 😔)